fdbserver/fdbserver.actor.cpp

```diff
@@ -216,7 +216,8 @@ CSimpleOpt::SOption g_rgOptions[] = {
 	{ OPT_NEW_CLUSTER_KEY,       "--new-cluster-key",           SO_REQ_SEP },
 	{ OPT_AUTHZ_PUBLIC_KEY_FILE, "--authorization-public-key-file", SO_REQ_SEP },
 	{ OPT_KMS_CONN_DISCOVERY_URL_FILE,           "--discover-kms-conn-url-file",            SO_REQ_SEP },
-	{ OPT_KMS_CONNECTOR_TYPE,    "--kms-connector-type",        SO_REQ_SEP },
+	{ OPT_KMS_CONNECTOR_TYPE,    "--kms-connector-type",                                    SO_REQ_SEP },
+	{ OPT_KMS_REST_ENABLE_NOT_SECURE_CONECTION,  "--kms-rest-enable_not_secure_connection", SO_REQ_SEP },
```

Comment:
shall we config it only via the knob and not a command-line argument? More ways to config it means more things we need to check to validate the config of a cluster.

Reply:
If we do keep this, we should use hyphens rather than underscores.

Reply:
@sfc-gh-yiwu I think it is better to be command-line-arguments to make it more explicit. 
@sfc-gh-ajbeamon thanks for pointing out, shall do the needful.

Thanks

Reply:
I feel like this could be `SO_NONE`. If this option is present we will enable unsecure connection, otherwise we will not

Reply:
@sfc-gh-ahusain do you know if there's way to check the knob value outside of FDB? If no and there's two ways to config the knob, we have two thing to check in the `foundationdb.conf` for validation. Both command-line argument and knobs are settable via `foundationdb.conf`.

Reply:
@sfc-gh-yiwu if I understood the question/concern here? 

My understanding introducing Encryption cluster configuration is, command-line-arguments are a way to take inputs from the user and for FDB to internally use it, one needs to define an underlying KNOB.

Reply:
@sfc-gh-ahusain I get what you mean, but to me since knobs are also settable via `--knob_xxx` command-line argument, the knobs are also pretty user-facing. I'm okay with adding the command-line argument. Just a matter of preference.

Reply:
@sfc-gh-yiwu I think the idea is SOMEDAY, the command line arguments can be stored as IMMUTABLE_KNOBS, for which there isn't any support today :)